bin/jwt.sh

```diff
@@ -0,0 +1,57 @@
+#/bin/bash
+
+# Helper script to handle JWT (JSON Web Tokens)
+# https://jwt.io/
+#
+# Depends on jq, openssl and bc
+
+if [ -z $AMO_USER ]; then
+  echo "$0 expects AMO_USER env variable to be set"
+  exit
+fi
+if [ -z $AMO_SECRET ]; then
+  echo "$0 expects AMO_SECRET env variable to be set"
+  exit
+fi
+
+set -euo pipefail
+IFS=$'\n\t'
+
+HEADER='{
+  "alg": "HS256",
+  "typ": "JWT"
+}'
+PAYLOAD='{
+  "iss": "'$AMO_USER'",
+  "jti": '$(echo "scale=10; $RANDOM*10/32767" | bc)',
+  "iat": '$(date +%s)',
+  "exp": '$(($(date +%s)+60))'
+}'
+
+function base64_encode()
+{
+  declare INPUT=${1:-$(</dev/stdin)};
+  echo -n "$INPUT" | openssl enc -base64 | tr '+\/' '-_' | tr -d '=' | tr -d '\r\n'
+}
+
+# For some reason, probably bash-related, JSON that terminates with an integer
+# must be compacted. So it must be something like `{"userId":1}` or else the
+# signing gets screwed up. Weird, but using `jq -c` works to fix that.
+function json() {
```

Comment:
the purpose of this method is only to compact the json right? so json_compact() ?